Textile/Classes/FeedItemData.h

```diff
@@ -0,0 +1,43 @@
+//
+//  FeedItemData.h
+//  Textile
+//
+//  Created by Aaron Sutula on 6/17/19.
+//
+
+#import <Foundation/Foundation.h>
+
+@class Text;
+@class Comment;
+@class Like;
+@class Files;
+@class Ignore;
+@class Join;
+@class Leave;
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef enum : NSUInteger {
+  FeedItemTypeText,
+  FeedItemTypeComment,
+  FeedItemTypeLike,
+  FeedItemTypeFiles,
+  FeedItemTypeIgnore,
+  FeedItemTypeJoin,
+  FeedItemTypeLeave
+} FeedItemType;
+
+@interface FeedItemData : NSObject
+
+@property (nonatomic, assign) FeedItemType type;
```

Comment:
The idea is to switch on this `type` property, and then the corresponding property below will have a values while all the others will be `nil`.